### ISSUE TITLE
FETCH_ERROR "TypeError [ERR_INVALID_URL]: Invalid URL" for requests made in tests (after migration from `fetch` to RTK Query)

### ISSUE THREAD

**Issue by naXa777** [kettanaito/msw#1597]
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 14 or higher

### Node.js version

19.3

### Reproduction repository

https://github.com/OKitel/react2023Q1/tree/9387604ef84c981f0a8e840b9723aebff821bdc8/react-playground

### Reproduction steps

```
cd react-playground
npm i
npm run test:ui
```

### Current behavior

tests fail with [FETCH_ERROR "TypeError [ERR_INVALID_URL]: Invalid URL" for requests made in tests](https://stackoverflow.com/q/76046546/1429387)

### Expected behavior

tests pass. fake data is fetched from the mock server.

**Comment by frisch-raphael** [comment]
Got a similar problem while trying to use useFetch from vueuse

**Comment by mymactive** [comment]
I encountered a similar issue in my environment.
I managed to find a workaround.
By switching your test environment from "happy-dom" to another environment such as 'jsdom', this problem can be resolved.

```typescript
/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';

// https://vitejs.dev/config/
export default defineConfig({
plugins: [react()],
test: {
environment: 'jsdom', // <- change from 'happy-dom'
});
```

This solution worked in my case.
I suspect that the cause may be linked to the URL implementation in happy-dom.

**Comment by mscottnelson** [comment]
I ran into this in a nodejs context and dug into it a bit. In my case, this was occurring because msw 1.2.3 is using an old version of mswjs/interceptors (0.17.9) that incorrectly defines the `fetch` interface. In particular, the call to `fetch` in that version attempts to determine the URL of the fetch with

```js
const url = typeof input === 'string' ? input : input.url
```

where `input` is the first argument to `fetch`.
However, `fetch` accepts type `URL | RequestInfo` as its first argument, and URL does not have a `url` property, so the url was being set to `undefined`.

This problem is fixed in newer versions of mswjs/interceptors, but these newer versions are not available in `main`. In my case, the problem is fixed by switching over to msw@next, but this may not be an option for many since there are many breaking changes. If it's the same problem for others, you may be able to side-step the problem by making sure to use type `string | RequestInfo` with any calls to `fetch`.

It appears you can follow this issue here: https://github.com/mswjs/msw/pull/1436, which, from reading the comments, looks like it will hit `main` soon. If you want to see how much you'll need to change to adopt that newer version (ie, msw@next, for the moment), there is more information at https://github.com/mswjs/msw/blob/feat/standard-api/MIGRATING.md

**Comment by satanshiro** [comment]
I can confirm that this also works for me when upgrading msw to the release candidate 0.0.0-fetch.rc-17, the changes look like a complete overhaul and i'm not sure its close to being finished,
what is the release schedule? did anyone find another solution?

**Comment by mscottnelson** [comment]
@satanshiro I don't actually know the answer to your question as I'm just a user like yourself, but judging from https://github.com/mswjs/msw/pull/1436#issuecomment-1666833227, this appears to be very close to complete, with most breaking changes in place, and the one remaining change dealing with exception handling.

**Comment by kettanaito** [comment]
This has been fixed as a part of #1999.
